client/me/purchases/upcoming-renewals/upcoming-renewals-dialog.tsx

```diff
@@ -0,0 +1,190 @@
+			args: { expiry: moment( purchase.expiryDate ).fromNow() },
+		} );
+	}
+	return translate( 'expires %(expiry)s', {
+		comment:
+			'"expiry" is relative to the present time and it is already localized, eg. "in a year", "in a month", "a week ago"',
+		args: {
+			expiry: moment( purchase.expiryDate ).fromNow(),
+		},
+	} );
+}
+
+const UpcomingRenewalsDialog: FunctionComponent< Props > = ( {
+	site,
+	purchases,
+	isVisible,
+	onClose,
+	onConfirm,
+} ) => {
+	const translate = useTranslate();
+	const moment = useLocalizedMoment();
+	const [ selectedPurchases, setSelectedPurchases ] = useState< number[] >( [] );
+
+	const purchasesSortByRecentExpiryDate = useMemo(
+		() => [ ...purchases ].sort( ( a, b ) => a?.expiryDate?.localeCompare( b?.expiryDate ) ),
+		[ purchases ]
+	);
+
+	useEffect( () => {
+		if ( isVisible ) {
+			setSelectedPurchases( purchases.map( ( purchase ) => purchase.id ) );
+		}
+	}, [ isVisible, purchases ] );
+
+	const confirmSelectedPurchases = useCallback( () => {
+		onConfirm( purchases.filter( ( purchase ) => selectedPurchases.includes( purchase.id ) ) );
+	}, [ purchases, selectedPurchases, onConfirm ] );
+
+	return (
+		<Dialog
+			isVisible={ isVisible }
+			leaveTimeout={ 0 }
+			additionalClassNames="upcoming-renewals-dialog"
+			onClose={ onClose }
+		>
+			<h2 className="upcoming-renewals-dialog__header">{ translate( 'Upcoming renewals' ) }</h2>
+			<h3 className="upcoming-renewals-dialog__subheader">
+				{ translate( 'Site: %(siteName)s', { args: { siteName: site.domain } } ) }
+			</h3>
+			<hr />
+			{ purchasesSortByRecentExpiryDate.map( ( purchase ) => {
+				const expiresText = getExpiresText( translate, moment, purchase ) as string;
+				const purchaseTypeText = purchaseType( purchase );
+				const onChange = () => {
+					if ( selectedPurchases.includes( purchase.id ) ) {
+						setSelectedPurchases( selectedPurchases.filter( ( id ) => id !== purchase.id ) );
+					} else {
+						setSelectedPurchases( selectedPurchases.concat( [ purchase.id ] ) );
+					}
+				};
+				return (
+					<Fragment key={ purchase.id }>
+						<div className="upcoming-renewals-dialog__row">
+							<FormLabel
+								optional={ false }
+								required={ false }
+								className="upcoming-renewals-dialog__label"
+							>
+								<div className="upcoming-renewals-dialog__checkbox">
+									<FormInputCheckbox
+										className="upcoming-renewals-dialog__checkbox-input"
+										name={ `${ purchase.productSlug }-${ purchase.id }` }
+										checked={ selectedPurchases.includes( purchase.id ) }
+										onChange={ onChange }
+									/>
+								</div>
+								<div className="upcoming-renewals-dialog__name">
+									{ getName( purchase ) }
+									<div className="upcoming-renewals-dialog__detail">
+										{ purchaseTypeText ? `${ purchaseTypeText }: ` : '' }
+										<span className={ isExpired( purchase ) ? 'expired' : '' }>
+											{ purchaseTypeText ? expiresText : capitalize( expiresText ) }
```

Comment:
Do you know if `capitalize()` works well in all languages?  (I guess even if it doesn't, it's probably not that big of a deal to have this wind up uncapitalized.)

Reply:
It internally uses the native `toUpperCase` method so it will be as safe as that method is, it'll probably won't work perfectly in some cases but it shouldn't represent a problem.